### ISSUE TITLE
Error When Running image_train.py

### ISSUE THREAD

**Issue by hwan-sig** [forever208/ADM-ES#3]
Description:
I attempted to run the following command to train the model:

`OPENAI_LOGDIR='./Logs' PYTHONPATH='.' CUDA_VISIBLE_DEVICES=0 python scripts/image_train.py --optimizer adamw --image_size 32 --num_channels 128 --num_res_blocks 3 --diffusion_steps 1000 --noise_schedule cosine --lr 1e-4 --batch_size 128 --learn_sigma True --eps_scaler=0 --lr_anneal_steps 100000 &`

However, the script produced an error as shown in the attached screenshot.
Could you help identify the cause of this issue and provide guidance on how to resolve it? Thank you!

<img width="286" alt="스크린샷 2024-12-30 오후 2 11 37" src="https://github.com/user-attachments/assets/c88c5723-1a27-403f-b722-7ae7f4672597" />

**Comment by forever208** [comment]
@hwan-sig  it is very likely that the training dataset is not clean, which dataset are you training on?

**Comment by hwan-sig** [comment]
I used CIFAR10 dataset.
When I perform the code without `--learn_sigma True`, it works well (It denotes all loss, mse, and vb values)
However, with  `--learn_sigma True`, it causes the error. Thank you.!

**Comment by forever208** [comment]
@hwan-sig it is a bit weird the problem happens on CIFAR-10. Does the nan occur immediately at the beginning of the training? Can you also print out the loss of `eps` and `vlb` (when using --learn_sigma) for debugging, and see if the loss scale is unusual? Also, consider to re-download the dataset

**Comment by hwan-sig** [comment]
I think I solved it.
I set eps_scaler to be 0 in training time and set eps_scaler to be 1 in sampling time.

**Comment by forever208** [comment]
@hwan-sig Thanks for letting me know, I just realized that it was a bug and have fixed it with a new commit
